src/backend/PostgresBackend.cpp

```diff
@@ -393,153 +450,99 @@ PostgresBackend::doFetchSuccessorKey(
 
 std::vector<TransactionAndMetadata>
 PostgresBackend::fetchTransactions(
-    std::vector<ripple::uint256> const& hashes) const
+    std::vector<ripple::uint256> const& hashes,
+    boost::asio::yield_context& yield) const
 {
+    if (!hashes.size())
+        return {};
+
     std::vector<TransactionAndMetadata> results;
```

Comment:
If we're anticipating that `results` will contain roughly `hashes.size()` entries, consider increasing the vector's capacity with `reserve`.